packages/core/lib/run-before-send.js

```diff
@@ -8,11 +8,13 @@ module.exports = (report, onError) => (fn, cb) => {
       if (ret && typeof ret.then === 'function') {
         return ret.then(
           // resolve
-          val => cb(null, shouldPreventSend(report, val)),
+          val => setTimeout(() => cb(null, shouldPreventSend(report, val)), 0),
```

Comment:
This ensures that any errors that are thrown in the callback `cb()` do not get swallowed by this Promise, and spat out as an unhandled rejection event.

By ensuring it executes outside of the synchronous stack of this promise call, it will throw an uncaught error.

The way this was manifesting itself was that an error in the tests was causing the tests to hang, rather than to fail the test with an error.